### PR TITLE
Refactor download pdf

### PR DESCRIPTION
**Pull request by fmassouNTTDATA**
this PR to resolve download file Error( on test 19, 14_1, 97, 98, 99, 100 ) and to resolve Error related to version of selenium and the new chrome version 115

### CODE REVIEW

File: src/main/java/it/pn/frontend/e2e/common/DownloadFile.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            double TotalDownload = 0.00;
+            int readbyte = 0; //Stores the number of bytes written in each iteration.
+            double percentOfDownload = 0.00;
+
+            URL url = new URL(urlLink);
+            HttpURLConnection http = (HttpURLConnection)url.openConnection();
+            double filesize = (double)http.getContentLengthLong();
+
+            BufferedInputStream input = new BufferedInputStream(http.getInputStream());
+            //fileLoc.createNewFile();
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/common/DownloadFile.java]: delete comment

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/mittente/DownloadFileMittentePagoPATest.java [hunk 2]
```diff
@@ -52,46 +53,71 @@ public void downloadECheckFile() {
...
+        int numeroLinkAvvenutaRicezione = dettaglioNotificaSection.getLinkAvvenutaRicezione();
+        for (int i = 1; i <numeroLinkAvvenutaRicezione ; i++) {
+            dettaglioNotificaSection.clickLinkAvvenutaRicezione(i);
+            try {
+                TimeUnit.SECONDS.sleep(5);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+            String urlAvvenutaRicezione = getUrl("https://webapi.test.notifichedigitali.it/delivery-push/"+ this.datiNotifica.get("codiceIUN").toString() +"/document/AAR?documentId=safestorage:");
+            file = new File(workingDirectory+"/src/test/resources/dataPopulation/downloadFileNotifica/mittente/notificaN"+count+".pdf");
```
Comment (leleOFA) [inline src/test/java/it/pn/frontend/e2e/stepDefinitions/mittente/DownloadFileMittentePagoPATest.java]: use final variable for root string

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/mittente/DownloadFileMittentePagoPATest.java [hunk 3]
```diff
@@ -52,46 +53,71 @@ public void downloadECheckFile() {
...
-            String urlFileAttestazioneOppponubile = getUrl("https://webapi.test.notifichedigitali.it/delivery-push/"+this.datiNotifica.get("codiceIUN").toString()+"/legal-facts/");
-            dettaglioNotificaSection.downloadFileNotifica("src/test/resources/dataPopulation/downloadFileNotifica/mittente",urlFileAttestazioneOppponubile,3);
+            String urlFileAttestazioneOppponubile = getUrl("https://webapi.test.notifichedigitali.it/delivery-push/");
+            file = new File(workingDirectory+"/src/test/resources/dataPopulation/downloadFileNotifica/mittente/notificaN"+count+".pdf");
+            count = count+1;
+            downloadFile.download(urlFileAttestazioneOppponubile,file);
         }
-
+        count = count -1;
+        downloadFile.controlloDownload(workingDirectory+"/src/test/resources/dataPopulation/downloadFileNotifica/mittente",count);
```
Comment (leleOFA) [inline src/test/java/it/pn/frontend/e2e/stepDefinitions/mittente/DownloadFileMittentePagoPATest.java]: use final variable for root string

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/mittente/DownloadFileMittentePagoPATest.java (hunk 2 above)
Comment (fmassouNTTDATA) [reply src/test/java/it/pn/frontend/e2e/stepDefinitions/mittente/DownloadFileMittentePagoPATest.java]: done

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/mittente/DownloadFileMittentePagoPATest.java (hunk 3 above)
Comment (fmassouNTTDATA) [reply src/test/java/it/pn/frontend/e2e/stepDefinitions/mittente/DownloadFileMittentePagoPATest.java]: done

File: src/main/java/it/pn/frontend/e2e/common/DownloadFile.java (hunk 1 above)
Comment (fmassouNTTDATA) [reply src/main/java/it/pn/frontend/e2e/common/DownloadFile.java]: done

**Review by leleOFA — APPROVED** [review]

**Pull request merged**